packages/react-router/modules/matchPath.js

```diff
@@ -44,14 +47,19 @@ const matchPath = (pathname, path, options = {}) => {
   if (exact && !isExact)
     return null
 
+  const params = Object.assign({},
+    parent && parent.params,
+    keys.reduce((memo, key, index) => {
```

Comment:
Mentioned above, but this merges parent `params` and the match's `params` into one object. I know that in v2/3, there were two different params objects, so I'm not sure if it would be preferred to return one object with only this `path`'s params and one with both params merged.